### PR TITLE
Add error when attempting to export GSSP page

### PR DESCRIPTION
**Pull request by ijjk**
As discussed this adds an error when a user tries to export a page with `getServerSideProps`

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| buildDuration | 9.4s | 9.2s | -210ms |
| nodeModulesSize | 54.9 MB | 54.9 MB | ⚠️ +1.16 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..8eb0.js gzip | 9.78 kB | 9.78 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.4 kB | 55.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.77 kB | 4.77 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.71 kB | 6.71 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| _app.js gzip | 1.09 kB | 1.09 kB | ✓ |
| _error.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.39 kB | 7.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| _app.module.js gzip | 594 B | 594 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.28 kB | 5.28 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| index.html gzip | 916 B | 916 B | ✓ |
| link.html gzip | 925 B | 925 B | ✓ |
| withRouter.html gzip | 914 B | 914 B | ✓ |
| Overall change | 2.75 kB | 2.75 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| buildDuration | 10s | 10s | ⚠️ +7ms |
| nodeModulesSize | 54.9 MB | 54.9 MB | ⚠️ +1.16 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..8eb0.js gzip | 9.78 kB | 9.78 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.4 kB | 55.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.77 kB | 4.77 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.71 kB | 6.71 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| _app.js gzip | 1.09 kB | 1.09 kB | ✓ |
| _error.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.39 kB | 7.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| _app.module.js gzip | 594 B | 594 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.28 kB | 5.28 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/NEXT-145 | Change |
| - | - | - | - |
| _error.js gzip | 293 kB | 294 kB | ⚠️ +776 B |
| 404.html gzip | 1.32 kB | 1.32 kB | ✓ |
| hooks.html gzip | 957 B | 957 B | ✓ |
| index.js gzip | 293 kB | 294 kB | ⚠️ +265 B |
| link.js gzip | 301 kB | 301 kB | ⚠️ +1 B |
| routerDirect.js gzip | 299 kB | 300 kB | ⚠️ +416 B |
| withRouter.js gzip | 300 kB | 300 kB | ⚠️ +19 B |
| Overall change | 1.49 MB | 1.49 MB | ⚠️ +1.48 kB |

</details>

</details>

File: errors/gssp-export.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# getServerSideProps Export Error
+
+#### Why This Error Occurred
+
+You attempted to export a page with `getServerSideProps` which is not allowed. `getServerSideProps` is meant for requesting up to date information on every request which means exporting it defeats the purpose of the method.
+
+#### Possible Ways to Fix It
+
+If you would like the page be static you can leverage the new `getStaticProps` method instead.
```
Comment (timneutkens) [inline errors/gssp-export.md]:
```suggestion
If you would like the page be static you can leverage the `getStaticProps` method instead.
```

File: errors/gssp-export.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# getServerSideProps Export Error
+
+#### Why This Error Occurred
+
+You attempted to export a page with `getServerSideProps` which is not allowed. `getServerSideProps` is meant for requesting up to date information on every request which means exporting it defeats the purpose of the method.
+
+#### Possible Ways to Fix It
+
+If you would like the page be static you can leverage the `getStaticProps` method instead.
+
```
Comment (timneutkens) [inline errors/gssp-export.md]:
```suggestion
```

File: errors/gssp-export.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+If you are attempting to export specific pages and didn't mean to export a page with `getServerSideProps` you can leverage `exportPathMap` in your `next.config.js` to export the specific pages.
```
Comment (timneutkens) [inline errors/gssp-export.md]:
```suggestion
```

**Pull request merged**